woocommerce/woocommerce
# Do not set WOOCOMMERCE_BLOCKS_PHASE inside build-zip script.

### Changes proposed in this Pull Request:

The `bin/build-zip.sh` script is not always run in a prod setting so we should not set the `WOOCOMMERCE_BLOCKS_PHASE` env variable inside the script unless it has not been set already. Now in the place we need to override it (live branch workflow) we pass in the override value so experimental blocks can be built.

In future this variable should be removed, but for now keeping mention of it in the script is an ok interim step.


### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. You can check if experimental blocks are available in the live branch for this PR. Create a JN site from the PR, go to block editor and try add an experimental block like "Collection Filter". You can also check the contents of the blocks.ini file that is bundled with the plugin to see the value is correct.
2. Testing the other builds will require that you fork this repo, setup actions and run this. Or if you feel more comfortable with the simplicity of these changes, you could simulate a local test of the change by running `WOOCOMMERCE_BLOCKS_PHASE=1 ./bin/build-zip.sh` and see that the resulting zip has the correct blocks.ini value.



### Changelog entry




<details>

#### Significance




#### Type




#### Message

Do not set WOOCOMMERCE_BLOCKS_PHASE in the build-zip.sh script, pass it when calling the script in relevant workflows.

#### Comment 

</details>
